From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tobias Koch <tobias.koch@aeltra.eu>
Date: Sat, 8 Aug 2026 19:38:58 +0300
Subject: [PATCH] libfetch: pass an unsigned char to isspace() in
 http_new_chunk()

The <ctype.h> functions are defined for values representable as unsigned
char and for EOF; passing a negative value is undefined.  *p has type
char, so on a platform where char is signed any byte >= 0x80 after the
chunk size is passed as a negative value.

Every other ctype call in libfetch already casts to unsigned char; this
one was missed.

Signed-off-by: Tobias Koch <tobias.koch@aeltra.eu>
---
 libfetch/http.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libfetch/http.c b/libfetch/http.c
index bac6478..9acbda3 100644
--- a/libfetch/http.c
+++ b/libfetch/http.c
@@ -142,7 +142,7 @@ http_new_chunk(struct httpio *io)
 		return -1;
 
 	io->chunksize = fetch_parseuint(io->conn->buf, &p, 16, SIZE_MAX);
-	if (*p && *p != ';' && !isspace(*p))
+	if (*p && *p != ';' && !isspace((unsigned char)*p))
 		return -1;
 
 	return 0;
